azext_iot/central/providers/device_provider.py

```diff
@@ -181,8 +181,8 @@ def get_device_registration_info(
         if info:
             return info
 
-        device = self.get_device(device_id)
-        if device.device_status == DeviceStatus.provisioned:
+        device = self.get_device(device_id, central_dns_suffix)
+        if DeviceStatus(device.device_status.value) == DeviceStatus.provisioned:
```

Comment:
You don't need to grab the value and recast, the `device_status` is expected to be the enum itself

Reply:
i.e `device.device_status == DeviceStatus.provisioned`